docs/src/main/tut/kleisli.md

```diff
@@ -0,0 +1,237 @@
+---
+layout: default
+title:  "Kleisli"
+section: "data"
+source: "https://github.com/non/cats/blob/master/core/src/main/scala/cats/data/Kleisli.scala"
+scaladoc: "#cats.data.Kleisli"
+---
+# Kleisli
+Kleisli is a data type that will come in handy often, especially if you are working with monadic functions.
+
+## Functions
+One of the most useful properties of functions is that functions **compose**. That is, given a function
+`A => B` and a function `B => C`, we can compose them to create a new function `A => C`. It is through
+this compositional property that we are able to write many small functions and compose them together
+to create a larger one that suits our needs.
+
+Often times, our functions will return monadic values. For instance, consider the following set of functions.
+
+```tut
+val parse: String => Option[Int] = s =>
+  try {
+    Some(s.toInt)
+  } catch {
+    case _: NumberFormatException => None
+  }
+
+val reciprocal: Int => Option[Double] =
+  i => if (i != 0) Some(1.0 / i) else None
+```
+
+As it stands we cannot use `Function1.compose` (or `Function1.andThen`) to compose these two functions.
+The output type of `parse` is `Option[Int]` whereas the input type of `reciprocal` is `Int`.
+
+This is where `Kleisli` comes into play.
+
+## Kleisli
+At it's core, `Kleisli[F[_], A, B]` is just a wrapper around the function `A => F[B]`. Depending on the
+properties of the `F[_]`, we can do different things with `Kleisli`s. For instance, if `F[_]` has a
+`FlatMap[F]` instance (e.g. is equipped with a `flatMap: F[A] => (A => F[B]) => F[B]` function), we can
```

Comment:
I think the `.flatMap` type signature there is a bit confusing. Would it be enough to say something like "(i.e. we can call `flatMap` on `F[A]` values)"?